File: 04-polimorfismo/python/rectangulo.py
from dibujable import Dibujable
from figura import Figura


class Rectangulo(Figura, Dibujable):

    def __init__(self, ancho, alto):
        super().__init__("Rectangulo")
        self._ancho = ancho
        self._alto = alto

    def area(self):
        return self._ancho * self._alto

    def perimetro(self):
        return 2 * (self._ancho + self._alto)

    def dibujar(self):
        print(f"Dibujando un rectangulo {self._ancho}x{self._alto}  [ ]")

File: 04-polimorfismo/java/Rectangulo.java
public class Rectangulo extends Figura implements Dibujable {

    private double ancho, alto;

    public Rectangulo(double ancho, double alto) {
        super("Rectangulo");
        this.ancho = ancho;
        this.alto = alto;
    }

    @Override
    public double area() {
        return ancho * alto;
    }

    @Override
    public double perimetro() {
        return 2 * (ancho + alto);
    }

    @Override
    public void dibujar() {
        System.out.println("Dibujando un rectangulo " + ancho + "x" + alto + "  [ ]");
    }
}
